OpenRA.Mods.Common/Activities/DeployForGrantedCondition.cs

```diff
@@ -0,0 +1,69 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Activities
+{
+	public class DeployForGrantedCondition : Activity
+	{
+		readonly GrantConditionOnDeploy deploy;
+		readonly int facing;
+		readonly bool canTurn;
+
+		public DeployForGrantedCondition(Actor self) : base()
+		{
+			canTurn = self.Info.HasTraitInfo<IFacingInfo>();
+			facing = self.Info.TraitInfo<GrantConditionOnDeployInfo>().Facing;
+			deploy = self.Trait<GrantConditionOnDeploy>();
```

Comment:
Making `GrantConditionOnDeploy.info` => `GrantConditionOnDeploy.Info` public would save a trait info lookup here and GrantConditionOnDeploy passing itself would save the trait lookup here (and would allow conditional deploy traits).
```C#
		public DeployForGrantedCondition(Actor self, GrantConditionOnDeploy deploy)
		{
			canTurn = self.Info.HasTraitInfo<IFacingInfo>();
			this.deploy = deploy;
			facing = deploy.Info.Facing;
```